File: ABC/ABC301-400/ABC338/E/accumulate/E.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
	int N;
	cin >> N;
	vector<int> A(N), B(N), cntA(2 * N), cntB(2 * N);
	for (int i = 0; i < N; i++) {
		cin >> A[i] >> B[i];
		A[i]--; B[i]--;
		if (A[i] > B[i]) {
			swap(A[i], B[i]);
		}
		cntA[A[i]]++;
		cntB[B[i]]++;
	}
	for (int i = 1; i < 2 * N; i++) {
		cntA[i] += cntA[i - 1];
		cntB[i] += cntB[i - 1];
	}
	for (int i = 0; i < N; i++) {
		if (cntB[B[i] - 1] - cntB[A[i]] < cntA[B[i] - 1] - cntA[A[i]]) {
			cout << "Yes\n";
			return 0;
		}
	}
	cout << "No\n";
}

File: ABC/ABC301-400/ABC338/E/accumulate/E.nim
import strutils, sequtils

let N = stdin.readLine.parseInt
var
  A = newSeq[int](N)
  B = newSeq[int](N)
  cntA = newSeq[int](2 * N)
  cntB = newSeq[int](2 * N)
for i in 0..<N:
  let a_b = stdin.readLine.split().map(parseInt)
  A[i] = a_b[0] - 1
  B[i] = a_b[1] - 1
  if A[i] > B[i]:
    swap(A[i], B[i])
  inc cntA[A[i]]
  inc cntB[B[i]]
for i in 1..<(2 * N):
  cntA[i] += cntA[i - 1]
  cntB[i] += cntB[i - 1]
for i in 0..<N:
  if cntB[B[i] - 1] - cntB[A[i]] < cntA[B[i] - 1] - cntA[A[i]]:
    echo "Yes"
    quit()
echo "No"